lib/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,109 @@
+import React, {
+  Component,
+  ReactNode,
+  AllHTMLAttributes,
+  Fragment,
+} from 'react';
+import ThemeConsumer from '../ThemeConsumer/ThemeConsumer';
+import Box from '../Box/Box';
+import FieldLabel from '../FieldLabel/FieldLabel';
+import FieldMessage from '../FieldMessage/FieldMessage';
+import classnames from 'classnames';
+import styles from './TextField.css.js';
+
+type InputProps = AllHTMLAttributes<HTMLInputElement>;
+type RequiredInputProps = 'id' | 'value' | 'onChange';
+interface TextFieldProps
+  extends Required<Pick<InputProps, RequiredInputProps>> {
+  label?: string;
+  secondaryLabel?: ReactNode;
+  tertiaryLabel?: ReactNode;
+  placeholder?: string;
+  message?: ReactNode | false;
+  tone?: 'neutral' | 'critical' | 'positive';
+  type?:
```

Comment:
Personally I think we should lock this down to text and maybe search? All the others will have some form of custom UI on top and are probably separate components. Thoughts?

Reply:
Yeah I flipped a bit on this, at a minimum I think we should have: text, password, email, search, number, tell and url. Thoughts?